### PR TITLE
dont error the entire repair process when a repair step errors

### PR DESCRIPTION
**Pull request by icewind1991**
Instead show the error as any other repair step error

Also includes formatting changes to keep the errors from being drowned out by the rest of the repair output

### CODE REVIEW

**Review by kesselb — APPROVED** [review]

File: lib/private/Repair.php [hunk 1]
```diff
@@ -113,7 +113,11 @@ public function run() {
 		foreach ($this->repairSteps as $step) {
 			$this->currentStep = $step->getName();
 			$this->emit('\OC\Repair', 'step', [$this->currentStep]);
-			$step->run($this);
+			try {
+				$step->run($this);
+			} catch (\Exception $e) {
+				$this->emit('\OC\Repair', 'error', [$e->getMessage()]);
```
Comment (MorrisJobke) [inline lib/private/Repair.php:119]: I would love more details here. At least in the logs.

Comment (rullzer) [reply lib/private/Repair.php:119]: ping @icewind1991

Comment (icewind1991) [reply lib/private/Repair.php:124]: done

**Comment by rullzer** [comment]
Master is Nextcloud 22 now.
If this should go into 21 it should be backported.

**Comment by MorrisJobke** [comment]
Rebased

**Review by MorrisJobke — APPROVED** [review]
Code looks good 👍

**Pull request merged**